benchmark/index.js

```diff
@@ -421,6 +424,87 @@ test('private box2', (t) => {
   )
 })
 
+test('private box2 group keys', (t) => {
+  const sbot = SecretStack({ appKey: caps.shs })
+    .use(require('../'))
+    .use(require('../encryption-formats/box2'))
+    .call(null, {
+      keys: keys5,
+      path: dirBox2Group,
+    })
+
+  // variables
+
+  // all your groups keys  
+  const groups = 100
+
+  // for messages, the percentage of groups that includes you
+  const percentMe = 30
+
+  const otherKey = Buffer.from(
+    '12345d8f9cbf37f6d7062826f6decac93e308060a8aaaa77e6a4747f40ee1a76',
+    'hex'
+  )
+  sbot.box2.addGroupKey('other', otherKey)
+
+  for (let i = 0; i < groups; ++i)
+  {
+    const startOfKey = String(i).padStart(5, '0')
+
+    const testkey = Buffer.from(
+      startOfKey + 'd8f9cbf37f6d7062826f6decac93e308060a8aaaa77e6a4747f40ee1a76',
+      'hex'
+    )
+    sbot.box2.addGroupKey('mygroup' + i, testkey)
+  }
+
+  t.pass(`Group keys: ${groups}, percentage for me: ${percentMe}%`)
+
+  let contents = []
+  for (var i = 0; i < 1000; ++i) {
+    const percentage = Math.floor(Math.random() * 100)
+    const groupId = Math.floor(Math.random() * groups)
+    const group = percentage < percentMe ? 'mygroup' + groupId : 'other'
+
+    // group must be first
+    const recps = [group]
+    contents.push({ type: 'tick', count: i, recps })
+  }
+
+  startMeasure(t, 'add 1000 box2 group msgs')
+  pull(
+    pull.values(contents),
+    pull.asyncMap(sbot.db.publish),
```

Comment:
To make sure that no decryption happens on `sbot.db.publish` (who knows if one day we change the internals of db2), we could do `sbot.db.create` instead. And instead of passing `contents`, we pass `opts`, where `opts.push({ keys: someOtherKeys, content: {type:'tick', count: i, recps}})`

Reply:
right, can fix those later today. I was copy pasting the one above :-) So will fix that one as well

Reply:
oh ho ho, there was a bug in this code before. This is why box2 was "faster". The box2 test actually created box1 messages :horse: